### PR TITLE
migrate to kubelet config file

### PR DESCRIPTION
**Pull request by rphillips**
Migrates the kubelet to a kubelet.conf file.

The customer typically had to set configuration settings within the kubelet unit file and pass command-line flags to reconfigure the Kubelet. With this patch, the configuration file will allow a subset of options to be set by the customer.

### CODE REVIEW

File: templates/_base/worker/units/kubelet.yaml [hunk 1]
```diff
@@ -7,32 +7,26 @@ contents: |
...
         --allow-privileged \
         --node-labels=node-role.kubernetes.io/worker \
         --minimum-container-ttl-duration=6m0s \
-        --cluster-dns={{.ClusterDNSIP}} \
-        --cluster-domain=cluster.local \
         --client-ca-file=/etc/kubernetes/ca.crt \
         --cloud-provider={{cloudProvider .}} \
-        {{.CloudProviderConfig}} \
+        {{.CloudProviderConfig -}} \
+         \
```
Comment (abhinavdahiya) [inline templates/_base/worker/units/kubelet.yaml]: why this extra line ?

Comment (rphillips) [reply templates/_base/worker/units/kubelet.yaml]: fixed in 0ee9e75.

**Comment by abhinavdahiya** [comment]
/approve

@aaronlevy do you mid taking a quick look

**Comment by aaronlevy** [comment]
To my naive eyes this lgtm. I'm assuming flags that are left over are not represented in the config file?

Might want someone else from pod team / someone more familiar with kubelet mechanics to take a pass - but could always be follow up.

Not strictly related to this, but could also drop `--exit-on-lock-contention` flag -- I really really hope no one is still trying to self-host the kubelet (we might want to get that deprecated/removed upstream -- and I'll apologize to everyone for asking for it in the first place...)

**Comment by rphillips** [comment]
Removed the `--exit-on-lock-contention`. The rest of the flags have not been migrated into the kubelet configuration upstream yet.

**Comment by aaronlevy** [comment]
One last question is that it looks like `serialize-image-pulls` changed from false to true. Just confirming that this is intentional / expected?

**Comment by rphillips** [comment]
@aaronlevy good catch.

fixed and squashed.

**Comment by aaronlevy** [comment]
This has been dragging on (apologies) so I can lgtm, but also just noticed that `--lock-file=/var/run/lock/kubelet.lock` also still exists - which is also part of the self-hosting kubelet behavior we no longer use.

**Comment by rphillips** [comment]
@aaronlevy updated to remove the lock

**Pull request merged**